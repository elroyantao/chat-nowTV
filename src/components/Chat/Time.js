import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class Time extends Component {
  static propTypes = {
    timestamp: PropTypes.string.isRequired
  }
  render() {
    const { timestamp } = this.props

    const date = new Date(timestamp)
    const day = date.toLocaleString('en-GB',{
      hour12: true
    })
    return (
      <div className="Time">
        {day}
      </div>
    )
  }
}