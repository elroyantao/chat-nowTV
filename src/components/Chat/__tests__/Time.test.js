import testHelper from '../../../helpers/test-helper'
import Time from '../Time'

describe('<Time/>', () => {
  const renderComponent = testHelper(Time)
  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent({
        timestamp: '2016-11-24T15:49:20Z'
      }).getTree()).toMatchSnapshot()
    })
  })
})