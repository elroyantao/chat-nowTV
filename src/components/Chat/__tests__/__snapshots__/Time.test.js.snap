exports[`<Time/> @renders in default state 1`] = `
<div
  className="Time">
  11/24/2016, 3:49:20 PM
</div>
`;
